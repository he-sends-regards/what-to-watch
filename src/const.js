export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  ADD_REVIEW: `/films/:id/review`,
  FILM: `/films`,
  PLAYER: `/player`
};

export const APIRoute = {
  FILMS: `/films`,
  LOGIN: `/login`,
  PROMO_FILM: `/films/promo`,
  REVIEWS: `/comments`,
  FAVORITE: `/favorite`
};

export const HttpCode = {
  UNAUTHORIZED: 401
};

export const FilmRateDescriptions = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
  UNEXPECTED_RATE: `Unexpected rate`
};

export const FilmDescriptionTabsNames = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

export const ALL_GENRE = `All genres`;

export const FavoriteFlags = {
  ADD_TO_FAVORITE: 1,
  REMOVE_FROM_FAVORITE: 0
};
