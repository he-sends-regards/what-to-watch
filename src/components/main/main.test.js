import React from "react";
import renderer from "react-test-renderer";
import Main from "./main";
import configureStore from "redux-mock-store";
import mockAuthInfo from "../../mocks/authData";

import mockFilms from "../../mocks/films";
import mockPromoFilm from "../../mocks/promoFilm";
import {Provider} from "react-redux";
import {ALL_GENRE, AuthorizationStatus} from "../../const";
import {Router} from "react-router-dom";
import browserHistory from "../../browser-history";

describe(``, () => {
  const mockStore = configureStore([]);

  const store = mockStore({
    FILMS: {
      films: mockFilms,
      promoFilm: mockPromoFilm,
      activeGenre: ALL_GENRE,
      favoriteFilms: mockFilms,
      showFilmsCount: 0
    },
    USER: {
      authorizationStatus: AuthorizationStatus.NO_AUTH,
      authInfo: mockAuthInfo
    }
  });

  const mainComponent = renderer.create(
      <Router history={browserHistory}>
        <Provider store={store}>

          <Main/>
        </Provider>
      </Router>
  );

  it(`Should Main render correctly`, () => {
    expect(mainComponent.toJSON()).toMatchSnapshot();
  });
});
