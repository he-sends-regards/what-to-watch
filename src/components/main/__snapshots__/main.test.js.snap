// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Main render correctly 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Bronson"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          className="user-block__link"
          href="/login"
          onClick={[Function]}
          style={
            Object {
              "borderRadius": 0,
              "color": "inherit",
              "textDecoration": "none",
            }
          }
        >
          Sign In
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Bronson poster"
            height="327"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Bronson
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Action
            </span>
            <span
              className="movie-card__year"
            >
              2008
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <a
                href="/player/1"
                onClick={[Function]}
                style={
                  Object {
                    "color": "inherit",
                    "textDecoration": "none",
                  }
                }
              >
                <svg
                  height="19"
                  viewBox="0 0 19 19"
                  width="19"
                >
                  <use
                    xlinkHref="#play-s"
                  />
                </svg>
                <span>
                  Play
                </span>
              </a>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="14"
                viewBox="0 0 18 14"
                width="18"
              >
                <use
                  xlinkHref="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            ALL_GENRE
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Action
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      />
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          onClick={[Function]}
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
